specs/builder.go

```diff
@@ -0,0 +1,312 @@
+				IDLength:    dummyField.IDLength,
+				Description: dummyField.Description,
+				Enc:         enc,
+				Pref:        pref,
+				Pad:         pad,
+			})
+		case reflect.TypeOf(field.Binary{}).String():
+			newField = field.NewBinary(&field.Spec{
+				Length:      dummyField.Length,
+				IDLength:    dummyField.IDLength,
+				Description: dummyField.Description,
+				Enc:         enc,
+				Pref:        pref,
+				Pad:         pad,
+			})
+		case reflect.TypeOf(field.Bitmap{}).String():
+			newField = field.NewBitmap(&field.Spec{
+				Length:      dummyField.Length,
+				IDLength:    dummyField.IDLength,
+				Description: dummyField.Description,
+				Enc:         enc,
+				Pref:        pref,
+				Pad:         pad,
+			})
+		default:
+			continue
+		}
+
+		spec.Fields[index] = newField
+	}
+
+	return &spec, nil
+}
+
+func (builder *messageSpecBuilder) ExportJSON(orgSpec *iso8583.MessageSpec) ([]byte, error) {
+
+	if orgSpec == nil {
+		return nil, fmt.Errorf("invalid message spec")
+	}
+
+	dummyJson := specDummy{
+		Name:   orgSpec.Name,
+		Fields: make(map[string]fieldDummy),
+	}
+
+	for index, orgField := range orgSpec.Fields {
+		dummyName := fmt.Sprintf("Field%03d", index)
+		dummyField := fieldDummy{
+			Type: reflect.TypeOf(orgField).Elem().String(),
+		}
+
+		switch reflect.TypeOf(orgField).Elem() {
+		case reflect.TypeOf(field.String{}), reflect.TypeOf(field.Numeric{}), reflect.TypeOf(field.Binary{}), reflect.TypeOf(field.Bitmap{}):
+			spec := orgField.Spec()
+
+			dummyField.Length = spec.Length
+			dummyField.IDLength = spec.IDLength
+			dummyField.Description = spec.Description
+			if spec.Enc != nil {
+				dummyField.Enc = reflect.TypeOf(spec.Enc).Elem().String()
+			}
+			if spec.Pref != nil {
+				dummyField.Prefix = reflect.TypeOf(spec.Pref).Elem().String()
+
+				lengthStr := ""
+				dummy := ""
+				fmt.Sscanf(spec.Pref.Inspect(), "%s %s %s", &dummy, &lengthStr, &dummy)
+				if lengthStr != "" && lengthStr != "fixed" && strings.ToUpper(lengthStr) == strings.Repeat("L", len(lengthStr)) {
+					dummyField.PrefixLength = len(lengthStr)
+				}
+			}
+			if spec.Pad != nil {
+				pad := padDummy{}
+				switch reflect.TypeOf(spec.Pad).Elem().String() {
+				case "padding.leftPadder":
+					if spec.Pad.Inspect() != nil {
+						pad.Type = reflect.TypeOf(spec.Pad).Elem().String()
+						pad.Pad, _ = utf8.DecodeRune(spec.Pad.Inspect())
+					}
+				}
+				if pad.Type != "" {
+					dummyField.Padding = &pad
+				}
+			}
+
+		default:
+			continue
+		}
+
+		dummyJson.Fields[dummyName] = dummyField
+	}
+
+	outputBuffer := new(bytes.Buffer)
+	enc := json.NewEncoder(outputBuffer)
+	enc.SetEscapeHTML(false)
+	enc.SetIndent("", "\t")
+	err := enc.Encode(dummyJson)
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to export message spec")
+	}
+
+	return outputBuffer.Bytes(), nil
+}
+
+func getPrefixInterface(prefixName string, length int) prefix.Prefixer {
+
+	prefixName = "*" + prefixName
+	var pref prefix.Prefixer
+
+	switch prefixName {
```

Comment:
technically, you can just create a map:

```
var prefixes = map[string]prefix.Prefixer{
  "ASCII.Fixed": prefix.ASCII.Fixed,
  "ASCII.L": prefix.ASCII.L,
  "ASCII.LL": prefix.ASCII.LL,
  ...
}
```

then we don't need to care about prefix lenght.

Reply:
fixed